### DISCUSSION TITLE
Request for Online Workshop: Programming WebAssembly in Rust

### DISCUSSION THREAD

**Discussion by c12i** [RustaceansKenya/Events#3] [General]
Greetings,

I am writing to request an online workshop focused on programming WebAssembly (Wasm) in Rust. Given the growing importance of WebAssembly in modern web development and Rust's strong position in this ecosystem, I believe such a workshop would be extremely valuable for our community.

I would like to propose the following topics to be covered in the workshop:

1. Foundational WebAssembly / Introduction
   - Overview of WebAssembly and its importance
   - How Wasm fits into the web ecosystem

2. Writing WebAssembly Modules
   - Introduction to WebAssembly Text Format (WAT)
   - Creating Wasm modules with Rust
   - Interfacing Wasm modules with JavaScript

3. Rust-Wasm Tooling
   - Using wasm-pack for Rust and Wasm integration
   - Exploring wasm-bindgen for Rust and JavaScript interoperability
   - Working with web_sys and js_sys crates
   - Introduction to the gloo library

4. Building a Micro UI Framework with Rust
   - Hands-on session on creating a simple UI framework using Rust and Wasm

5. Established UI Frameworks
   - Overview of popular Rust-based UI frameworks for Wasm:
     - Dioxus
     - Leptos
     - Yew

6. WebAssembly Outside the Browser
   - Exploring Wasm's applications beyond web browsers
   - Use cases in serverless computing, plugins, and more

The workshop could be structured as a series of sessions, allowing for in-depth exploration of each topic. I believe this would provide a comprehensive introduction to WebAssembly development with Rust, catering to both beginners and more experienced developers looking to expand their skills.

I look forward to hearing your thoughts on this proposal.

**Comment by 448-OG** [comment]
I support this initiative. @nyambura00 @kiarie404 can help in planning this.

**Comment by kiarie404** [comment]
> I look forward to hearing your thoughts on this proposal.  

I think it's Nice and deep
The topics are well chosen and are likely to span days if not weeks.    

Just need some clarifications ...
Will this👇🏾 will be the major part of the workshop?
>     4. Building a Micro UI Framework with Rust
>        
>        * Hands-on session on creating a simple UI framework using Rust and Wasm

If it will be the major part, will you build the framework as you go or will it be [eltr](https://github.com/c12i/eltr)?

**Comment by nyambura00** [comment]
Hi @c12i !

Looking forward to hacking on WebAssembly. The topics cover a great range and would be highly valuable to the community.
